runtime/vm/ClassInitialization.cpp

```diff
@@ -444,9 +446,60 @@ classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitStat
 					}
 					iTable = iTable->next;
 				}
+				
+				clazz = VM_VMHelpers::currentClass(clazz);
+				
+				/* iterate over fields and load classes of fields marked flattenable and static*/
+				J9SRP *romValueTypeClasses = J9ROMCLASS_VALUETYPECLASSES(clazz->romClass);
+				J9Class **valueTypeClasses = clazz->valueTypeClasses;
+				UDATA valueTypeClassCount = clazz->romClass->valueTypeClassCount;
+				for (UDATA index = 0; (index < valueTypeClassCount) && (NULL != valueTypeClasses[index]); index++) {
+					if (J9_ARE_NO_BITS_SET((uintptr_t) valueTypeClasses[index], VALUE_TYPE_INSTANCE_FIELD_TAG)) {
+						J9UTF8 *signature = (J9UTF8 *) valueTypeClasses[index];
+						J9Class *currentClass = internalFindClassUTF8(currentThread, &J9UTF8_DATA(signature)[1], J9UTF8_LENGTH(signature)-2, clazz->classLoader, J9_FINDCLASS_FLAG_THROW_ON_FAIL);
+						if (J9_ARE_NO_BITS_SET(currentClass->romClass->modifiers, J9AccValueType)) {
+							J9BytecodeVerificationData *bcvd = vm->bytecodeVerificationData;
+							omrthread_monitor_enter(bcvd->verifierMutex);
+							bcvd->romClass = clazz->romClass;
+							bcvd->errorModule = J9NLS_BCV_ERR_FLATTENABLE_NOT_VALUE_TYPE__MODULE;
+							bcvd->errorCode = J9NLS_BCV_ERR_FLATTENABLE_NOT_VALUE_TYPE__ID;
+							U_8 *verifyErrorString = j9bcv_createVerifyErrorString(vm->portLibrary, bcvd);
+							omrthread_monitor_exit(bcvd->verifierMutex);
+							setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGINCOMPATIBLECLASSCHANGEERROR, (char *) verifyErrorString);
+							goto done;
```

Comment:
I used `j9bcv_createVerifyErrorString` like you suggested, but I'm not sure if I've implemented this entirely correctly - also the code looks a bit messy and is duplicated and I'm not really sure how to deal with it (or if its an issue) since making a helper function isn't super useful.

Reply:
just realized, `j9bcv_createVerifyErrorString` is not the right thing here. 

You can simply lookup the nls message and add it to `setCurrentExceptionUTF`

Reply:
i think you can use `j9nls_lookup_message` for the lookup

Reply:
I found that `setCurrentExceptionNLS` seems to be what we're looking for.